src/Symfony/Component/HttpKernel/HttpCache/ResponseCacheStrategy.php

```diff
@@ -64,33 +99,122 @@ public function update(Response $response)
             return;
         }
 
-        // Remove validation related headers in order to avoid browsers using
-        // their own cache, because some of the response content comes from
-        // at least one embedded response (which likely has a different caching strategy).
-        if ($response->isValidateable()) {
-            $response->setEtag(null);
-            $response->setLastModified(null);
+        // Remove validation related headers of the master response,
+        // because some of the response content comes from at least
+        // one embedded response (which likely has a different caching strategy).
+        $response->setEtag(null);
+        $response->setLastModified(null);
+
+        $this->add($response);
+
+        $response->headers->set('Age', $this->age);
+
+        if ($this->isNotCacheableResponseEmbedded) {
+            $response->setExpires($response->getDate());
+
+            if ($this->flagDirectives['no-store']) {
+                $response->headers->set('Cache-Control', 'no-cache, no-store, must-revalidate');
+            } else {
+                $response->headers->set('Cache-Control', 'no-cache, must-revalidate');
+            }
+
+            return;
+        }
+
+        $flags = array_filter($this->flagDirectives);
+
+        if (isset($flags['must-revalidate'])) {
+            $flags['no-cache'] = true;
         }
 
-        if (!$response->isFresh() || !$response->isCacheable()) {
-            $this->cacheable = false;
+        $response->headers->set('Cache-Control', implode(', ', array_keys($flags)));
+
+        $maxAge = null;
+        $sMaxage = null;
+
+        if ($this->ageDirectives['max-age']) {
+            $maxAge = $this->ageDirectives['max-age'] + $this->age;
```

Comment:
Well the value is that we know the age of the response, essentially when exactly the oldest "fragment" of this response was cached. Otherwise this information would be lost and every new request for the same URI would get a different response result (with updated max-age etc.).

Reply:
okay, i see the value of keeping that information stable.